ssl/statem/extensions.c

```diff
@@ -361,6 +365,49 @@ static int validate_context(SSL *s, unsigned int extctx, unsigned int thisctx)
     return 1;
 }
 
+int tls_validate_all_contexts(SSL *s, unsigned int thisctx, RAW_EXTENSION *exts)
+{
+    size_t i, num_exts, builtin_num = OSSL_NELEM(ext_defs), offset;
+    RAW_EXTENSION *thisext;
+    unsigned int context;
+    ENDPOINT role = ENDPOINT_BOTH;
+
+    if ((thisctx & SSL_EXT_CLIENT_HELLO) != 0)
+        role = ENDPOINT_SERVER;
+    else if ((thisctx & SSL_EXT_TLS1_2_SERVER_HELLO) != 0)
+        role = ENDPOINT_CLIENT;
+
+    /* Calculate the number of extensions in the extensions list */
+    num_exts = builtin_num + s->cert->custext.meths_count;
+
+    for (thisext = exts, i = 0; i < num_exts; i++, thisext++) {
+        if (!thisext->present)
+            continue;
+
+        if (i < builtin_num) {
+            context = ext_defs[i].context;
+        } else {
+            custom_ext_method *meth = NULL;
+
+            if ((thisctx & SSL_EXT_CLIENT_HELLO) != 0)
+                role = ENDPOINT_SERVER;
+            else if ((thisctx & SSL_EXT_TLS1_2_SERVER_HELLO) != 0)
+                role = ENDPOINT_CLIENT;
```

Comment:
Oops, the duplication inside the loop is unnecessary. I've deleted it. The idea was that this would be a generic function. Although there is currently only one call site there are still two possible contexts (`SSL_EXT_TLS1_2_SERVER_HELLO` and `SSL-EXT_TLS1_3_SERVER_HELLO`). The check for `SSL_EXT_CLIENT_HELLO` could be removed at the loss of the generic nature of the function...but, I kind of feel it is better to keep it generic.